src/IssueViz.Security/ObservingIssueLocationStore.cs

```diff
@@ -0,0 +1,123 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security
+{
+    internal interface IObservingIssueLocationStore : IIssueLocationStore, IDisposable
```

Comment:
Does this interface need to inherit from 'IIssueLocationStore'?

The consumers of the additional methods on this interface (the individual issue stores) don't need to know anything about the `IIssueLocationStore`, do they? From their point of view, they just interested in registering themselves with an issue store observer/sink - they don't need to know what the observer/sink does.

Something like `IIssueStoreObserver`, `IIssueStoreSink` instead?

Reply:
Good idea, I went with `IIssueStoreObserver`